test/LiquidityManager/invest.test.ts

```diff
@@ -152,165 +235,123 @@ describe.only('LiquidityManager#invest', () => {
-                {
-                    positionManager: positionManagerUniV3,
-                    inputToken: stablecoin,
-                    depositAmountInputToken: amount,
+        const { token0, token1 } = UniswapV3.sortTokens(
+            await mockTokenWithAddress(ETH_PRICE_BN, 18, weth),
+            await mockTokenWithAddress(BTC_PRICE_BN, 18, wbtc),
+        )
 
-                    fee: 3_000,
+        const {
+            amount0,
+            amount1,
+            tickLower,
+            tickUpper,
+            swapAmountToken0,
+            swapAmountToken1,
+        } = UniswapV3.getMintPositionInfo(
+            amountWithDeductedFees,
+            pool,
+            token0.price,
+            token1.price,
+            10, // 10% lower
+            10, // 10% upper
+        )
 
-                    token0,
-                    token1,
+        await investAndValidateTransaction(
+            pool,
+            amount0,
+            amount1,
+            token0,
+            token1,
+            swapAmountToken0,
+            swapAmountToken1,
+            tickLower,
+            tickUpper,
+        )
+    })
 
-                    swapToken0: wbtcIsToken0 ? wbtcEncodedSwap : wethEncodedSwap,
-                    swapToken1: wbtcIsToken0 ? wethEncodedSwap : wbtcEncodedSwap,
+    it('should add liquidity using a price range outside the current price', async () => {
+        const pool = await UniswapV3Helpers.getPoolByContract(stableBtcLpUniV3)
 
-                    swapAmountToken0: halfAmountWithDeductedFees,
-                    swapAmountToken1: halfAmountWithDeductedFees,
+        const { token0, token1 } = UniswapV3.sortTokens(
+            await mockTokenWithAddress(USD_PRICE_BN, 18, stablecoin),
+            await mockTokenWithAddress(BTC_PRICE_BN, 18, wbtc),
+        )
 
-                    // TODO calculate tick dinamically using price
-                    tickLower: getNearestUsableTick(tick - (tick / 10n), tickSpacing),
-                    tickUpper: getNearestUsableTick(tick + (tick / 10n), tickSpacing),
+        const {
+            amount0,
+            amount1,
+            tickLower,
+            tickUpper,
+            swapAmountToken0,
+            swapAmountToken1,
+        } = UniswapV3.getMintPositionInfo(
+            amountWithDeductedFees,
+            pool,
+            token0.price,
+            token1.price,
+            10, // 10% lower
+            -20, // 10% upper
+        )
 
-                    amount0Min: wbtcIsToken0 ? wbtcAmountMin : wethAmountMin,
-                    amount1Min: wbtcIsToken0 ? wethAmountMin : wbtcAmountMin,
-                },
-                permitAccount0,
-            )
+        await investAndValidateTransaction(
+            pool,
+            amount0,
+            amount1,
+            token0,
+            token1,
+            swapAmountToken0,
+            swapAmountToken1,
+            tickLower,
+            tickUpper,
+        )
```

Comment:
you should add an `expect` statement to validate if the position is indeed 100% in one token since it is out of range